csi/controller.go

```diff
@@ -1300,6 +1326,70 @@ func (s *OsdCsiServer) listMultipleSnapshots(
 	return listSnapshotsResp, nil
 }
 
+// ListSnapshots lists all cloud backups as snapshot in a cluster.
+func (s *OsdCsiServer) listCloudSnapshots(
+	ctx context.Context,
+	req *csi.ListSnapshotsRequest,
+) (*csi.ListSnapshotsResponse, error) {
+	cloudBackupClient, err := s.getCloudBackupClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	credentialID := req.GetSecrets()[osdSnapshotCredentialIDKey]
+
+	maxBackups := req.GetMaxEntries()
+	if len(req.GetSnapshotId()) > 0 {
+		maxBackups = 1
+	}
+	resp, err := cloudBackupClient.EnumerateWithFilters(ctx, &api.SdkCloudBackupEnumerateWithFiltersRequest{
+		SrcVolumeId:       req.GetSourceVolumeId(),
+		CloudBackupId:     req.GetSnapshotId(),
+		CredentialId:      credentialID,
+		ContinuationToken: req.GetStartingToken(),
+		MaxBackups:        uint64(maxBackups),
+	})
+	if nil != err {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to list cloud backups: %v", err)
+	}
+
+	csiSnapshotResp := &csi.ListSnapshotsResponse{
+		Entries: make([]*csi.ListSnapshotsResponse_Entry, len(resp.Backups)),
+	}
+
+	for i, backup := range resp.Backups {
+		// Check if snapshot has been created but is in error state
+		backupStatus, errFindFailed := cloudBackupClient.Status(ctx, &api.SdkCloudBackupStatusRequest{
```

Comment:
EnumerateWithFilters returns the status as part of the response:
https://github.com/libopenstorage/openstorage/blob/master/api/api.proto#L4727

Can we remove this per-backup status call and instead depend on the single enumerate with filters response?